docs/architecture/adr-004.md

```diff
@@ -0,0 +1,55 @@
+
+
+## Context
+
+With the bank module, a cosmos-chain is able to produce and manage multi-asset coin. Later, with IBC, it became also possible for a cosmos chain to transfer coin-asset to another chain and also manage asset originated from another chain. All coin-assets in Cosmos follow the same standard and are managed by the bank module.
+We are calling them "native tokens"
+
+On the other side, the evm module introduce the notion of ethereum virtual machine. Like in Ethereum, it is possible with smart contract to create all form of assets. The erc20 standard is the most popular standard to describe fungible tokens and are integrated in most web3 tools (metamask, explorer etc).
+
+This ADR describes a mechanism to convert seamlessly native tokens to evm tokens (erc20) out of an IBC transfer. In a more general concept, any kind of native tokens can be converted to an evm tokens (and vice versa) using cronos native transactions. 
+
+In our reasoning, most of our users comes from the Ethereum world and are more familiar to interact with ethereum web3 tools for signing transactions and managing assets. This is why this proposal is important for the chain adoption.
+
+
+## Proposal
+
+- Implement in x/cronos module two cosmos transactions : `convert-vouchers` allows to convert an ibc-originated asset to its erc20 representation and `transfer-tokens` allows to send a erc20 to another cosmos chain through ibc.
+- Implement logic in `ibc-hooks` that allows to trigger automatically a `convert-vouchers` to asset originating from an ibc transfer.
+- Implement logic in `evm-hooks` that allows to trigger an IBC transfer from a contract account when a specific event is fired from the contract.
+
+
+## EVM 
+
+To be able to control native tokens with a smart contract, we are defining specific instruction that can be called by a module within a smart contract :
+
+- mint_by_cronos_module(address user, uint256 amount)
+- burn_by_cronos_module(address user, uint256 amount)
+- native_denom() returns (string)
+ 
+We are also defining specific events that can trigger specific action from smart contract's account on cosmos side
+-  __CronosSendToAccount(address recipient, uint256 amount) : send a specific token amount from contract address to recipient address
+-  __CronosSendToIBCChannel(address recipient, uint256 amount) : send a specific token amount through ibc to recipient address in destination chain
+
+## Further Discussions
+
+<!-- While an ADR is in the DRAFT or PROPOSED stage, this section should contain a summary of issues to be solved in future iterations (usually referencing comments from a pull-request discussion).
+Later, this section can optionally list ideas or improvements the author or reviewers found during the analysis of this ADR. -->
+
+
```

Comment:
also good to add consequences + link this ADR: https://github.com/crypto-org-chain/cronos/blob/main/docs/architecture/adr-002.md as it talks about it too